homeassistant/components/bouncie/sensor.py

```diff
@@ -0,0 +1,203 @@
+        )
+        sensor_types.append(
+            SensorEntityDescription(
+                key=f"car-speed-{vehicle[const.VEHICLE_VIN_KEY]}",
+                icon="mdi:speedometer",
+                name=f"{vehicle[const.VEHICLE_NICKNAME_KEY]} Speed",
+                device_class="speed",
+            )
+        )
+        sensor_types.append(
+            SensorEntityDescription(
+                key=f"car-mil-{vehicle[const.VEHICLE_VIN_KEY]}",
+                icon="mdi:engine",
+                name=f"{vehicle[const.VEHICLE_NICKNAME_KEY]} MIL",
+            )
+        )
+        sensor_types.append(
+            SensorEntityDescription(
+                key=f"car-battery-{vehicle[const.VEHICLE_VIN_KEY]}",
+                icon="mdi:car-battery",
+                name=f"{vehicle[const.VEHICLE_NICKNAME_KEY]} Battery",
+                device_class="",
+            )
+        )
+
+    entities = [
+        BouncieSensor(coordinator=coordinator, description=description)
+        for description in sensor_types
+    ]
+
+    async_add_entities(entities, False)
+
+
+class BouncieSensor(CoordinatorEntity[BouncieDataUpdateCoordinator], SensorEntity):
+    """Bouncie sensor."""
+
+    _attr_attribution = ATTRIBUTION
+
+    def __init__(
+        self,
+        coordinator: BouncieDataUpdateCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Init the BouncieSensor."""
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._state = None
+        self._attrs: dict[str, str] = {}
+        self._attr_name = f"{description.name}"
+        self._attr_unique_id = f"{coordinator._client_id}-{description.key.lower()}"
+        self._vehicle_info = next(
+            (
+                v
+                for v in self.coordinator.data["vehicles"]
+                if v["vin"] == self.entity_description.key.split("-")[-1]
+            ),
+            None,
+        )
+
+    def _update_car_info_state(self):
+        self._state = (
+            "Running" if self._vehicle_info["stats"]["isRunning"] else "Not Running"
+        )
+
+    def _update_car_info_attributes(self):
+        self._attrs[const.ATTR_VEHICLE_MODEL_MAKE_KEY] = self._vehicle_info[
+            const.VEHICLE_MODEL_KEY
+        ]["make"]
+        self._attrs[const.ATTR_VEHICLE_MODEL_NAME_KEY] = self._vehicle_info[
+            const.VEHICLE_MODEL_KEY
+        ]["name"]
+        self._attrs[const.ATTR_VEHICLE_MODEL_YEAR_KEY] = self._vehicle_info[
+            const.VEHICLE_MODEL_KEY
+        ]["year"]
+        self._attrs[const.ATTR_VEHICLE_NICKNAME_KEY] = self._vehicle_info["nickName"]
+        self._attrs[const.ATTR_VEHICLE_STANDARD_ENGINE_KEY] = self._vehicle_info[
+            "standardEngine"
+        ]
+        self._attrs[const.ATTR_VEHICLE_VIN_KEY] = self._vehicle_info["vin"]
+        self._attrs[const.ATTR_VEHICLE_IMEI_KEY] = self._vehicle_info["imei"]
+        self._update_car_stats_attributes()
```

Comment:
You should use `device_info` to group your entities under one device and remove redundant infos from attrs.

Reply:
Can you share some example on how to use `device_info` (like you did for `tuple...` below)?

Reply:
Yes, here it is:
- https://github.com/home-assistant/core/blob/e11917b7cbdd21a93424f801dce0be3802c8cb89/homeassistant/components/synology_dsm/entity.py#L53-L60
- https://github.com/home-assistant/core/blob/e11917b7cbdd21a93424f801dce0be3802c8cb89/homeassistant/components/renault/renault_vehicle.py#L51-L57 --> then used `self._attr_device_info = self.vehicle.device_info`
- https://github.com/home-assistant/core/blob/e11917b7cbdd21a93424f801dce0be3802c8cb89/homeassistant/components/freebox/router.py#L158-L168

Note: we now set device info with `self._attr_device_info` not with `@property`